src/CdrReader.ts

```diff
@@ -165,6 +171,78 @@ export class CdrReader {
     return value;
   }
 
+  /** Reads the delimiter header returning the endianness flag and object size
+   * NOTE: changing endian-ness with a single CDR message is not supported
+   */
+  dHeader(): { littleEndianFlag: boolean; objectSize: number } {
+    const header = this.uint32();
+    // DHEADER(O) = (E_FLAG<< 31) + O.ssize
+    /**
+     * E = 1 indicates that following the header XCDR stream
+     * endianness shall be changed to LITTLE_ENDIAN.
+     * E = 0 indicates that following the header XCDR stream
+     * endianness shall be changed to BIG_ENDIAN.
+     */
+    const littleEndianFlag = (header & 0x80000000) !== 0;
+    // We don't support changing the endianness mid stream, mostly because we don't have data to test it with
+    if (littleEndianFlag !== this.littleEndian) {
+      throw new Error(
+        "Dheader contained an endianness flag that did not match the stream. This is not supported at this time.",
+      );
+    }
+
+    const objectSize = header & 0x7fffffff;
+    return { littleEndianFlag, objectSize };
+  }
+
+  /**
+   * Reads the member header (EMHEADER) and returns the member ID, mustUnderstand flag, and object size
+   */
+  emHeader(): { mustUnderstand: boolean; id: number; objectSize: number } {
+    const header = this.uint32();
+    // EMHEADER = (M_FLAG<<31) + (LC<<28) + M.id
+    // M is the member of a structure
+    // M_FLAG is the value of the Must Understand option for the member
+    const mustUnderstand = Math.abs((header & 0x80000000) >> 31) === 1;
+    // LC is the value of the Length Code for the member.
+    const lengthCode = (header & 0x70000000) >> 28;
+    const id = header & 0x0fffffff;
+
+    const objectSize = this.emHeaderObjectSize(lengthCode);
+
+    return { mustUnderstand, id, objectSize };
+  }
+
+  /** Uses the length code to derive the member object size in
+   * the EMHEADER, sometimes reading NEXTINT (the next uint32
+   * following the header) from the buffer */
+  private emHeaderObjectSize(lengthCode: number) {
+    // 7.4.3.4.2 Member Header (EMHEADER), Length Code (LC) and NEXTINT
+    switch (lengthCode) {
+      case 0:
+        return 1;
+      case 1:
+        return 2;
+      case 2:
+        return 4;
+      case 3:
+        return 8;
+      // NEXTINT exists after header
+      case 4:
+        return this.uint32();
+      case 5:
+        return this.uint32();
```

Comment:
I don't understand case 5 :(

Reply:
It's just the same value 
<img width="520" alt="image" src="https://github.com/foxglove/cdr/assets/10187776/1d7fe06e-6422-40d1-94bd-a539a37874d3">
I'll put a comment though just to make it clearer it's not a mistake